### ISSUE TITLE
Owner as first-class citizen.

### ISSUE THREAD

**Issue by gavofyork** [ethereum/cpp-ethereum#684]
```
contract Foo: owned {
function Foo() { /*init*/ }
function kill() restrict {
  this.suicide(owner);
}
}
```

evaluates to

```
contract Foo {
function Foo() {
  owner = msg.sender;
  /*init*/
}
function kill() {
  if (msg.sender == owner) {
    this.suicide(owner);
  }
}
address owner;
}
```

**Comment by chriseth** [comment]
I would like to split this into two features:
1. restricted calls (given a condition which defaults to something given at contact level)
2. inheritance 
The default restriction condition is inherited.

**Comment by chriseth** [comment]
Implemented in #825 and #856.
